contrib/google.golang.org/grpc/grpc.go

```diff
@@ -31,11 +31,15 @@ func startSpanFromContext(ctx context.Context, method, operation, service string
 	return tracer.StartSpanFromContext(ctx, operation, opts...)
 }
 
-// withStreamError returns a tracer.WithError finish option, disregarding OK, EOF and Canceled errors.
-func withStreamError(err error) tracer.FinishOption {
+// finishWithStreamError applies finish option and a tag with gRPC status code, disregarding OK, EOF and Canceled errors.
+func finishWithStreamError(span ddtrace.Span, err error) {
 	errcode := status.Code(err)
-	if err == io.EOF || errcode == codes.Canceled || errcode == codes.OK || err == context.Canceled {
+	if err == io.EOF || err == context.Canceled {
```

Comment:
This is not what I meant, this is what this function should be:
```go
func finishWithError(span ddtrace.Span, err error) {
	errcode := status.Code(err)
	if err == io.EOF || errcode == codes.Canceled || errcode == codes.OK || err == context.Canceled {
             err = nil
        }
	span.SetTag(tagCode, errcode.String())
	span.Finish(tracer.WithError(err))
}
```
Do not change the code

Reply:
If here this keeps error as is, the error code will be always Unknown because the error is context.Cancled.
If we want this status code Unknown here, I will change. How do you think?

Reply:
Yep, that's fine. Users should always see the status code returned by gRPC. In this case, it's unknown, so that's correct.

Reply:
OK! I will do, thanks!

Reply:
Thank _you_!!!